orchestrator/lambdas/layers/heimdall_repos/heimdall_repos/bitbucket_utils.py

```diff
@@ -184,7 +185,27 @@ def _get_ref_names(self, repo: str, default: str, branch: str = None) -> Tuple[l
             for ref in repo_refs:
                 ref_name = self.service_helper.get_branch_name(ref)
                 ref_names.add(ref_name)
-                timestamps[ref_name] = ref["target"]["date"]
+
+                # Retrieve the timestamp of the latest commit
+                if "target" in ref and "date" in ref["target"]:
+                    timestamps[ref_name] = ref["target"]["date"]
+                else:
+                    # Query the commit endpoint for the timestamp of the latest commit
+                    commit_id = ref["latestCommit"]
+                    commit_url = self.service_helper.construct_bitbucket_commit_url(
+                        self.service_info.url, self.service_info.org, repo, commit_id
+                    )
+                    commit_query_response = self._query_bitbucket_api(commit_url)
+                    commit_query_response_dict = self.json_utils.get_json_from_response(commit_query_response)
+
+                    # Convert timestamp from milliseconds to seconds
+                    timestamp = commit_query_response_dict["committerTimestamp"]
+                    timestamp = timestamp / 1000
+
+                    # Format timestamp
+                    timestamp = time.strftime("%Y-%m-%dT%H:%M:%SZ", time.localtime(timestamp))
```

Comment:
Since we're formatting the timestamp in UTC (`Z`), we should use `time.gmtime()` instead of `localtime()`